File: testingtool.py
import re

#Validation for MIS
def valid_MIS(mis):
    try:
        i = mis.isdecimal()
        if i==True:
            print("Valid MIS")
        else:
            print("Invalid MIS")
    except:
        print("Error")

#Validation for Name
def valid_name(name):
    name = name.replace(" ","")
    try:
        i = name.isalpha()
        if i == True:
            print("Valid Name")
        else:
            print("Invalid name")
    except:
        print("Error)")

#Validation for phone number
def valid_phone(phone):
    try:
        if not len(phone) == 10:
            print("Invalid Phone number, does not have 10 digits")
        elif not phone.isdecimal():
            print("Invalid characters in phone number")
        else:
            print("Valid phone Number")
    except:
        print("Error")

#Validation for Blood Group        
def valid_bg(bg):
    try:
        if bg == "O+" or bg == "A+" or bg == "A-" or bg == "AB+" or bg == "AB-" or bg == "B+" or bg == "B-" or bg == "O-":
            print("Valid Blood Group")
        else:
            print("Invalid BloodGroup")
    except:
        print("Error")
#test cases
# valid_phone("12345") #invalid case
# valid_phone("123456789=") #invalid case
# valid_phone("9175801482") #valid case

# valid_MIS("111903134") #valid MIS

# valid_name("Anjali Dofe") #valid name
# valid_name("Aditya0444") #invalid name


  

File: testingtool.js
function validateName(name) {
    const re = /^[a-z ,.'-]+$/i;
    return re.test(String(name).toLowerCase());
}

function valid_name(name) {
	// var small = ['a','b','c', 'd', 'e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z',' '];
	// var up = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z',' '];
	for (let i = 0; i < name.length; i++) {
		if (name[i] < 65 || (name[i] > 91 && name[i] < 97) || name[i] >122 ){
			console.log("Invalid Name");
			return false;
		}
		else {
			console.log("Valid Name");
			return true;
		}
	}
}

function checkIfStringHasOnlyDigits(_string) {
    for (let i = _string.length - 1; i >= 0; i--) {
    const codeValue = _string.charCodeAt(i);
    if (codeValue < 48 || codeValue > 57) 
        return false
    }
    return true
}

function validateEmail(email) {
    const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(email).toLowerCase());
}

function valid_MIS(MIS) {
	if (MIS.length == 9 && checkIfStringHasOnlyDigits(MIS) == true){
        console.log("Valid MIS");
        return true;
    }
    else{
        console.log("Invalid MIS");
        return false;
    }
}

function valid_age(age) {
	int_age = Number(age)
    if (checkIfStringHasOnlyDigits(age) == false || int_age < 18 || int_age > 22){
        console.log("Invalid Age");
        return false;
    }
    else {
    	console.log("Valid Age");
    	return true;
    }
}

function valid_phone(phone) {
	if (checkIfStringHasOnlyDigits(phone) == false || phone.length != 10){
        console.log("Invalid Phone Number");
        return false;
    }
    else {
    	console.log("Valid Phone number");
    	return true;
    }
}

function valid_gradYead(year){
	int_year = Number(year)
    if (year.length > 0 && (int_year > 2025 || int_year < 2022)){
        console.log("Invalid Graduation Year");
        return false;
    }
    else {
    	console.log("Valid Graduation Year");
    	return true;
    }
}

function checkPassport(passportNum){
    const re = /^[A-Z]{1}[0-9]{7}$/;
    return re.test(String(passportNum));
}

function valid_bg(bloodGroup) {
	var bg = ['A+', 'A-', 'B+', 'B-', 'AB+', 'AB-', 'O+', 'O-'];
	 if (bg.includes(bloodGroup) == false){
        console.log("Invalid bloodGroup");
        return false;
    }
    else {
    	console.log("Valid Blood Group");
    	return true;
    }
}

if (valid_bg("A+") == false){
	console.log("FAILED TEST CASE")
}
else{
	console.log("TEST CASE PASSED")
}

console.log("\n")

if (valid_bg("AP") == false){
	console.log("FAILED TEST CASE")
}
else{
	console.log("TEST CASE PASSED")
}

console.log("\n")

if (valid_name("Anjali") == false){
	console.log("FAILED TEST CASE")
}
else {
	console.log("TEST CASE PASSED")
}

